src/main/java/org/apache/ibatis/builder/annotation/ProviderSqlSource.java

```diff
@@ -55,10 +55,17 @@ public ProviderSqlSource(Configuration configuration, Object provider) {
    * @since 3.4.5
    */
   public ProviderSqlSource(Configuration configuration, Object provider, Class<?> mapperType, Method mapperMethod) {
+    this(configuration, provider, mapperType, mapperMethod, configuration.getDefaultScriptingLanguageInstance());
+  }
+
+  /**
+   * @since 3.4.6
+   */
+  public ProviderSqlSource(Configuration configuration, Object provider, Class<?> mapperType, Method mapperMethod, LanguageDriver languageDriver) {
```

Comment:
It is possible for LanguageDriver to be configured via annotations, not by default.
```java
  private LanguageDriver getLanguageDriver(Method method) {
    Lang lang = method.getAnnotation(Lang.class);
    Class<? extends LanguageDriver> langClass = null;
    if (lang != null) {
      langClass = lang.value();
    }
    return assistant.getLanguageDriver(langClass);
  }
```

Reply:
I have modified it according to your suggestion.

```java
Lang lang = mapperMethod.getAnnotation(Lang.class);
      this.languageDriver = configuration.getLanguageRegistry().getDriver(lang == null ? configuration.getLanguageRegistry().getDefaultDriverClass() : lang.value());
```

Reply:
Almost. :)
Travis build failed due to a broken test (SqlProviderTest.notSqlProvider).
There needs to be a null check.

Reply:
Add NPE check for constructor parameter mapperMethod:
```java
Lang lang = mapperMethod == null ? null : mapperMethod.getAnnotation(Lang.class);
this.languageDriver = lang == null ?
    configuration.getLanguageRegistry().getDefaultDriver() :
    configuration.getLanguageRegistry().getDriver(lang.value());
```

Reply:
Thank you for your review.

Although this PR is inconspicuous, it will bring about a particularly big change.

After the PR merge, one of my open source projects will be improved for the current SNAPSHOT version, and MyBatis's generic DAO implementation will be easier.